pkg/apis/core/v1alpha1/uibutton_types.go

```diff
@@ -56,8 +57,16 @@ type UIButtonSpec struct {
 	// Location associates the button with another component for layout.
 	Location UIComponentLocation `json:"location" protobuf:"bytes,1,opt,name=location"`
 
-	// Text to appear on the button itself.
+	// Text to appear on the button itself or as hover text (depending on button location).
 	Text string `json:"text" protobuf:"bytes,2,opt,name=text"`
+
+	// Icon to appear next to button text or on the button itself (depending on button location).
+	//
+	// Valid values are icon font ligature names from the Material Icons set.
+	// See https://fonts.google.com/icons for the full list of available icons.
+	//
+	// +optional
+	Icon string `json:"icon,omitempty" protobuf:"bytes,3,opt,name=icon"`
```

Comment:
maybe call this IconName?

tbh, it's pretty fashionable among icon hipsters to embed SVGs everywhere. they're pretty small. so can we have IconSVG too?

Reply:
would probably call this 'IconName'

Reply:
I like the SVG idea - especially for the "help button" it might be nice to have a branded button, and shouldn't be too difficult on the frontend either.

(FYI To make the API friendlier to use, I didn't make it exclusive but documented SVG will take precedence if both are defined.)